dces-drc-integration/src/main/java/uk/gov/justice/laa/crime/dces/integration/service/ContributionService.java

```diff
@@ -54,13 +54,24 @@ public Integer processContributionUpdate(UpdateLogContributionRequest updateLogC
     @Timed(value = "laa_dces_drc_service_process_contributions_daily_files",
             description = "Time taken to process the daily contributions files from DRC and passing this for downstream processing.")
     public boolean processDailyFiles() {
-        Map<String, CONTRIBUTIONS> successfulContributions = new HashMap<>();
-        Map<String, String> failedContributions = new HashMap<>();
-        // get all the values to process via maat call
-        List<ConcurContribEntry> contributionsList = contributionClient.getContributions("ACTIVE");
-        sendContributionsToDrc(contributionsList, successfulContributions, failedContributions);
+        int defaultNoOfRecord = feature.noOfContributionRecords();
+        List<ConcurContribEntry> contributionsList;
+        int startingId = 0;
+        boolean isSuccessful = false;
+        do {
```

Comment:
While this is correct, I think that a `while (hasMoreRecords) {...}` loop would be more obvious to most developers.

`do {...} while (condition);` is not something people are used to see very often, and it might catch less experienced developers out.

You always starting from `startingdId = 0`, so could start with `hasMoreRecords = true;` and then at the end of the loop set `hasMoreRecords` based on the size of the `contributionList`. What do you think?